substrate/frame/revive/rpc/build.rs

```diff
@@ -0,0 +1,44 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+use std::process::Command;
+
+/// Get the current branch and commit hash.
+fn main() {
+	let output = Command::new("rustc")
+		.arg("--version")
+		.output()
+		.expect("cannot get the current rustc version");
+	// Exports the default rustc --version output:
+	// e.g. rustc 1.83.0 (90b35a623 2024-11-26)
+	// into the usual Ethereum web3_clientVersion format
+	// e.g. rustc1.83.0
+	let rustc_version = String::from_utf8_lossy(&output.stdout)
+		.split_whitespace()
+		.take(2)
+		.collect::<Vec<_>>()
+		.join("");
+	let target = std::env::var("TARGET").unwrap_or_else(|_| "unknown".to_string());
+
+	let repo = git2::Repository::open("../../../..").expect("should be a repository");
```

Comment:
I would be curious if this works when the crate is published and used in a dependency.

Reply:
Yea this will also 100% not work for `cargo vendor` builds.

Reply:
Actually this also does not build with `cargo remote`:  
`should be a repository: Error { code: -3, klass: 6, message: "could not find repository at '../../../..'" }`  
Please just put a default "unknown" version or something.

Reply:
Done here now https://github.com/paritytech/polkadot-sdk/pull/7786